src/history.ts

```
@@ -18,21 +18,26 @@ export class HistoryStore {
     };
 
     addElement(elem: HistoryElement) {
+        elem.internalTime = new Date().getTime();
         this.cropQueryElement(elem);
-        let currentHistory = this.getHistory();
+        let currentHistory = this.getHistory(false);
         if (currentHistory != null) {
             if (this.isValidEntry(elem)) {
                 this.setHistory([elem].concat(currentHistory));
             }
-
         } else {
             this.setHistory([elem]);
         }
     }
 
-    getHistory(): HistoryElement[] {
+    getHistory(stripInternalTime: boolean = true): HistoryElement[] {
         try {
-            return <HistoryElement[]> JSON.parse(this.store.getItem(STORE_KEY));
+            let history = <HistoryElement[]> JSON.parse(this.store.getItem(STORE_KEY));
+            if (stripInternalTime) {
+                return this.stripInternalTime(history);
```

If `this.stripInternalTime()` throws, it will be silently catched by the `catch` down there, then returning `[]` which would has high probability of causing headaches ;)

I would suggest wrapping `this.store.getItem(STORE_KEY)` in the try {} catch {} and leaving the rest out of it.

---

Done